File: projeto_2 - Kaella Sales/src/SeguroMoto.cpp
#include "../include/SeguroMoto.h"
#include <iostream>

SeguroMoto::SeguroMoto(Moto& moto, Cliente& cli, float v, std::string vi)
    : moto(moto), SeguroVeiculo(cli, v, vi) {}

void SeguroMoto::exibirDetalhes() {
    std::cout << "Seguro" << std::endl;
    std::cout << "Tipo: Motocicleta" << std::endl;
    std::cout << "Veículo: " << moto.getMarca() << " " << moto.getModelo() << " " << moto.getAno() << std::endl;
    std::cout << "Valor do seguro: " << valor << std::endl;
    std::cout << "Vigência: " << vigencia << "\n" << std::endl;
}

void SeguroMoto::CalcularValor(){
    valor += 275.0;
}

File: Projeto_2/projeto2_erlano/src/seguro/Seguro_moto.py
from src.seguro.Seguro_veiculo import SeguroVeiculo
from src.cliente import Cliente
from src.moto import Moto

class SeguroMoto(SeguroVeiculo):
    def __init__(self, cliente: Cliente, moto: Moto, valor: float, vigencia: str):
        super().__init__(cliente, valor, vigencia)
        self.moto = moto

    def calcular_valor(self):
        taxa_base = 0.07
        return self.valor * (1 + taxa_base)